### ISSUE TITLE
Java compatibility forced to match Gradle minimum Java compatability

### ISSUE THREAD

**Issue by MarkRx** [gradle-plugins/toolbox#95]
When specifying a minimum Gradle version the plugin forces the java compatibility to match the Gradle minimum compatibility version [here](https://github.com/gradle-plugins/toolbox/blob/v1.6.9/subprojects/gradle-plugin-development/src/main/java/dev/gradleplugins/internal/plugins/DefaultGradlePluginDevelopmentCompatibilityExtension.java#L49). This prevents plugins from building with a newer JDK because dependencies will fail to get pulled when running with newer versions of Gradle due to variant JDK version missmatches.

`- Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 8`

Users should be able to set compatibility to a version higher than the minimum Gradle java version.

**Comment by MarkRx** [comment]
It looks like this might have been [fixed](https://github.com/gradle-plugins/toolbox/blob/main/subprojects/gradle-plugin-development/src/main/java/dev/gradleplugins/internal/rules/WireMinimumGradleVersionWithJvmCompatibilityRule.java#L40) on the master branch. There does not appear to be a release yet with these changes though.

**Comment by lacasseio** [comment]
The new and old codes accounted for this by only forcing JVM compatibility if the user didn't specify anything. I checked, and it does work. However... I see the issue. Gradle automatically set the source compatibility when setting the target compatibility, which is what was left out. Could you confirm that you only set the target compatibility?

**Comment by lacasseio** [comment]
Version 1.6.13 should fix the issue. Please reopen if the problem is still present.

**Comment by TheMrMilchmann** [comment]
The fix in `1.6.13` actually broke the compilation in my case. Here is what I typically do:

```kotlin
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(22)
    }
}

withType<JavaCompile>().configureEach {
    options.release = 8
}
```

This configuration uses the latest toolchain to compile for a specific Java version. However, as of the changes in 1.6.13, this is no longer supported. The culprit seems to be that the compatibility versions are set to the toolchain versions regardless of whether `options.release` is specified and then take precedence.

For example, this issue surfaces when writing a plugin in Kotlin:

```kotlin
kotlin {
    compilerOptions {
        jvmTarget = JvmTarget.JVM_1_8
    }
}
```

Running a build then fails even if `options.release` is set with:

> Inconsistent JVM-target compatibility detected for tasks 'compileJava' (22) and 'compileKotlin' (1.8).

**Comment by TheMrMilchmann** [comment]
Opened #113 for visibility
